ccl/backupccl: TestDataDriven_external_connections_privileges failed

ccl/backupccl.TestDataDriven_external_connections_privileges [failed](https://mesolite.cluster.engflow.com/invocations/default/80fc4cb5-136e-4f6f-84df-924413547b85?testReportRun=1&testReportShard=19&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on release-24.1 @ [6cfb4f6a59e4437a1b281570b30090b12bd8d394](https://github.com/cockroachdb/cockroach/commits/6cfb4f6a59e4437a1b281570b30090b12bd8d394):


```
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/external-connections-privileges:23:
        exec-sql [1 args]
        CREATE EXTERNAL CONNECTION "testuser-ec" AS 'nodelocal://1/privs'
        ----
    datadriven.go:259: 
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/external-connections-privileges:27:
        exec-sql [0 args]
        CREATE TABLE foo (id INT);
        ----
    datadriven.go:259: 
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/external-connections-privileges:31:
        exec-sql [0 args]
        GRANT SELECT ON TABLE foo TO testuser
        ----
    datadriven.go:259: 
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/external-connections-privileges:36:
        exec-sql [1 args]
        BACKUP TABLE foo INTO 'external://testuser-ec'
        ----
        NOTICE: The existing privileges are being deprecated in favour of a fine-grained privilege model explained here https://www.cockroachlabs.com/docs/stable/backup.html#required-privileges. In a future release, to run BACKUP TABLE, user testuser will exclusively require the BACKUP privilege on tables: foo.
    datadriven.go:259: 
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/external-connections-privileges:41:
        exec-sql [0 args]
        GRANT USAGE ON EXTERNAL CONNECTION "testuser-ec" TO testuser;
        ----
    datadriven.go:259: 
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/external-connections-privileges:45:
        exec-sql [1 args]
        BACKUP TABLE foo INTO LATEST IN 'external://testuser-ec'
        ----
        NOTICE: The existing privileges are being deprecated in favour of a fine-grained privilege model explained here https://www.cockroachlabs.com/docs/stable/backup.html#required-privileges. In a future release, to run BACKUP TABLE, user testuser will exclusively require the BACKUP privilege on tables: foo.
    datadriven.go:259: 
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/external-connections-privileges:51:
        exec-sql [1 args]
        BACKUP TABLE foo INTO 'external://root'
        ----
        pq: user testuser does not have USAGE privilege on external_connection root
    datadriven.go:259: 
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/external-connections-privileges:56:
         SELECT * FROM system.privileges
        expected:
        root /externalconn/root {ALL} {} 1
        testuser /externalconn/testuser-ec {ALL} {} 100
        testuser /global/ {EXTERNALCONNECTION} {} 100
        
        found:
        root /externalconn/root {ALL} {} 1
        testuser /externalconn/testuser-ec {ALL} {} 101
        testuser /global/ {EXTERNALCONNECTION} {} 101
    --- FAIL: TestDataDriven_external_connections_privileges/backup-restore-privileges (0.74s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=19</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/test-eng
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven_external_connections_privileges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-38335

ccl/backupccl.TestDataDriven_external_connections_privileges [failed](https://tanzanite.cluster.engflow.com/invocations/default/bbcca887-7afa-4313-ab71-c0d1ddc49a26?testReportRun=18&testReportShard=19&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on release-24.1 @ [4571024fe5281a69eb1a1b302dd28c1ec0bd78c1](https://github.com/cockroachdb/cockroach/commits/4571024fe5281a69eb1a1b302dd28c1ec0bd78c1):


```
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/external-connections-privileges:23:
        exec-sql [1 args]
        CREATE EXTERNAL CONNECTION "testuser-ec" AS 'nodelocal://1/privs'
        ----
    datadriven.go:259: 
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/external-connections-privileges:27:
        exec-sql [0 args]
        CREATE TABLE foo (id INT);
        ----
    datadriven.go:259: 
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/external-connections-privileges:31:
        exec-sql [0 args]
        GRANT SELECT ON TABLE foo TO testuser
        ----
    datadriven.go:259: 
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/external-connections-privileges:36:
        exec-sql [1 args]
        BACKUP TABLE foo INTO 'external://testuser-ec'
        ----
        NOTICE: The existing privileges are being deprecated in favour of a fine-grained privilege model explained here https://www.cockroachlabs.com/docs/stable/backup.html#required-privileges. In a future release, to run BACKUP TABLE, user testuser will exclusively require the BACKUP privilege on tables: foo.
    datadriven.go:259: 
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/external-connections-privileges:41:
        exec-sql [0 args]
        GRANT USAGE ON EXTERNAL CONNECTION "testuser-ec" TO testuser;
        ----
    datadriven.go:259: 
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/external-connections-privileges:45:
        exec-sql [1 args]
        BACKUP TABLE foo INTO LATEST IN 'external://testuser-ec'
        ----
        NOTICE: The existing privileges are being deprecated in favour of a fine-grained privilege model explained here https://www.cockroachlabs.com/docs/stable/backup.html#required-privileges. In a future release, to run BACKUP TABLE, user testuser will exclusively require the BACKUP privilege on tables: foo.
    datadriven.go:259: 
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/external-connections-privileges:51:
        exec-sql [1 args]
        BACKUP TABLE foo INTO 'external://root'
        ----
        pq: user testuser does not have USAGE privilege on external_connection root
    datadriven.go:259: 
        /var/lib/engflow/worker/work/3/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/external-connections-privileges:56:
         SELECT * FROM system.privileges
        expected:
        root /externalconn/root {ALL} {} 1
        testuser /externalconn/testuser-ec {ALL} {} 100
        testuser /global/ {EXTERNALCONNECTION} {} 100
        
        found:
        root /externalconn/root {ALL} {} 1
        testuser /externalconn/testuser-ec {ALL} {} 101
        testuser /global/ {EXTERNALCONNECTION} {} 101
    --- FAIL: TestDataDriven_external_connections_privileges/backup-restore-privileges (1.03s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=18</code>
 - <code>shard=19</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven_external_connections_privileges.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/ccl/backupccl/testdata/backup-restore/external-connections-privileges
+++ b/pkg/ccl/backupccl/testdata/backup-restore/external-connections-privileges
@@ -53,26 +53,12 @@ BACKUP TABLE foo INTO 'external://root'
 ----
 pq: user testuser does not have USAGE privilege on external_connection root
 
-query-sql
-SELECT * FROM system.privileges
-----
-root /externalconn/root {ALL} {} 1
-testuser /externalconn/testuser-ec {ALL} {} 100
-testuser /global/ {EXTERNALCONNECTION} {} 100
-
 # Revoke the USAGE privilege. Note testuser had ALL privileges since they
 # created the External Connection, but revoking USAGE means that they will now
 # only have DROP privileges. Thus, they shouldn't be able to restore.
 exec-sql
 REVOKE USAGE ON EXTERNAL CONNECTION "testuser-ec" FROM testuser;
 ----
-
-query-sql
-SELECT * FROM system.privileges
-----
-root /externalconn/root {ALL} {} 1
-testuser /externalconn/testuser-ec {DROP} {} 100
-testuser /global/ {EXTERNALCONNECTION} {} 100
 
 exec-sql user=testuser
 RESTORE TABLE foo FROM LATEST IN 'external://testuser-ec'

